### PR TITLE
Chore api routes

### PR DESCRIPTION
**Pull request by codeMonkeyHopeful**
chores API routes applied

### CODE REVIEW

File: server/apiRoutes/ChoresRoutes.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+/*
+ * @ROUTE: POST to /api/chores/
+ * @DESC: allow admin to see all chores (if admin of any group), return nothing if not admin
+ * @ACCESS: admin only
+ */
+router.post('/', async (req, res, next) => {
+  const userId = req.body.userId;
+  let isAdmin = false;
```
Comment (HypeDis) [inline server/apiRoutes/ChoresRoutes.js:21]: why is isAdmin global?

File: server/apiRoutes/ChoresRoutes.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  const adminStatusByGroup = await findGroupInfo(userId);
+  //res.send(adminStatusByGroup);
+  for (let i = 0; i < adminStatusByGroup.length; i++) {
+    if (adminStatusByGroup[i].userIsAdmin) {
+      isAdmin = true;
+    }
+  }
+  if (isAdmin) {
+    const chores = await Chore.findAll({});
```
Comment (HypeDis) [inline server/apiRoutes/ChoresRoutes.js]: im having trouble understanding this. lets discuss tmr

File: server/apiRoutes/ChoresRoutes.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+const findGroupInfo = userId => {
+  return UserGroup.findAll({
+    where: { userId, userStatus: 'active' },
+  })
+    .then(result => result)
+    .catch(e => console.error(e));
+};
+
+/*
+ * @ROUTE: POST to /api/chores/
```
Comment (HypeDis) [inline server/apiRoutes/ChoresRoutes.js:15]: i think these first 2 routes belong in the group admin page.

File: server/apiRoutes/ChoresRoutes.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+ * @DESC: allow admin to see all chores (if admin of any group), return nothing if not admin
+ * @ACCESS: admin only
+ */
+router.post('/', async (req, res, next) => {
+  const userId = req.body.userId;
+  let isAdmin = false;
+  const adminStatusByGroup = await findGroupInfo(userId);
+  //res.send(adminStatusByGroup);
```
Comment (HypeDis) [inline server/apiRoutes/ChoresRoutes.js:23]: i think these first 2 routes belong in the group admin page.

File: server/apiRoutes/ChoresRoutes.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+ * @DESC: allow admin to see all assigned chores, return nothing if not admin
+ * @ACCESS: private
+ */
+router.post('/all_assigned_chores', async (req, res, next) => {
+  let chores;
+  const groups = [];
+  const userId = req.body.userId;
+  const allGroups = await findGroupInfo(userId);
+
+  //slim out groups to only those belonged to
```
Comment (HypeDis) [inline server/apiRoutes/ChoresRoutes.js:48]: i dont understand why this is needed

File: server/apiRoutes/ChoresRoutes.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    groups.push(allGroups[i].groupId);
+  }
+
+  //get all chores that are assigned in the group
+  for (let i = 0; i < groups.length; i++) {
+    const choreList = await Chore.findAll({
+      where: { groupId: groups[i] },
+      include: [{ model: AssignedChore }],
+    });
+    chores = choreList;
```
Comment (HypeDis) [inline server/apiRoutes/ChoresRoutes.js]: you are replacing chores each loop

File: server/apiRoutes/ChoresRoutes.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  //   }
+  // }
+});
+
+/*
+ * @ROUTE: POST to /api/chores/allpersonalchores
+ * @DESC: allow a user to see all their assigned chores
+ * @ACCESS: private
+ */
+router.post('/all_personal_chores', (req, res, next) => {
```
Comment (HypeDis) [inline server/apiRoutes/ChoresRoutes.js]: return an array of assigned chores organized by group

**Review by HypeDis — CHANGES_REQUESTED** [review]

File: server/apiRoutes/ChoresRoutes.js (hunk 7 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/ChoresRoutes.js]: updated to return object with groupId as key and array of assigned chores

File: server/apiRoutes/ChoresRoutes.js (hunk 6 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/ChoresRoutes.js]: modified data output to account for group ex. {groupId: [chore [assignedchores]]}

File: server/apiRoutes/ChoresRoutes.js (hunk 3 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/ChoresRoutes.js:15]: lets discuss tomorrow

File: server/apiRoutes/ChoresRoutes.js (hunk 1 above)
Comment (codeMonkeyHopeful) [reply server/apiRoutes/ChoresRoutes.js:21]: looping over potential multiple groups, need to set outside the loop in case it is changed later

**Pull request closed without merging**